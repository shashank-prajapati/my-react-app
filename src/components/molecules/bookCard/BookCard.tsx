import React, { useState } from "react";
import Card from "@mui/material/Card";
import BookDetails from "../bookDetails/BookDetails";
import { ButtonProps } from "@mui/material";
import Image from "../../atoms/image/Image";

interface Props extends ButtonProps {
  coverImage: string;
  bookTitle: string;
  bookAuthor: string;
  time: string;
  isButtoned: boolean;
  btnText?: string;
  handleClick?: any;
  index: number;
}

function BookCard(props: Props) {
  const {
    coverImage,
    bookTitle,
    bookAuthor,
    time,
    isButtoned,
    btnText,
    startIcon,
    index,
    handleClick,
  } = props;
  const [isHovered, setHovered] = useState(false);

  return (
    <div>
      <Card
        sx={{
          width: "19.7vw",
          backgroundColor: "secondary_100.main",
          borderRadius: "8px",
          pb: !isButtoned ? 4 : 0,
        }}
        data-testid="Card"
        onMouseEnter={() => setHovered(true)}
        onMouseLeave={() => setHovered(false)}
        onClick={() => handleClick && handleClick(index - 1)}
      >
        <Image source={coverImage} imgWidth="100%" imgHeight="100%" />
        <BookDetails
          bookTitle={bookTitle}
          bookAuthor={bookAuthor}
          time={time}
          isButtoned={isButtoned}
          btnText={btnText}
          startIcon={startIcon}
          isHovered={isHovered}
        />
      </Card>
    </div>
  );
}

export default BookCard;
