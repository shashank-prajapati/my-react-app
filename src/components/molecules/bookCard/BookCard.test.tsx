import * as React from "react";
import { render, fireEvent ,screen} from "@testing-library/react";
import BookCard from "./BookCard";
import { Typography } from "@mui/material";
import coverImage from "../../../media/coverImage.png";

describe("Testing Book Card", () => {
  it("should match the unButtoned Book Card Snapshot", () => {
    const { container } = render(
      <BookCard
        coverImage={coverImage}
        time="13"
        bookAuthor={`Jim Collins & Bill Lazier`}
        bookTitle="Beyond Entrepreneurship"
        isButtoned={false}
        index={0}
      ></BookCard>
    );
    expect(container).toMatchSnapshot();
  });
  it("should match the Buttoned Book Card Snapshot", () => {
    const { container } = render(
      <BookCard
        coverImage={coverImage}
        time="13"
        bookAuthor={`Jim Collins & Bill Lazier`}
        bookTitle="Beyond Entrepreneurship"
        isButtoned={true}
        btnText="Add to library"
        index={0}
      ></BookCard>
    );
    expect(container).toMatchSnapshot();
  });

  it("should call on click method of button", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <BookCard
        coverImage={coverImage}
        time="13"
        bookAuthor={`Jim Collins & Bill Lazier`}
        bookTitle="Beyond Entrepreneurship"
        isButtoned={true}
        btnText="Add to library"
        index={0}
        onClick={handleClick}
      ></BookCard>
    );
    fireEvent.click(screen.getByText("Add to library"));
    expect(handleClick).toHaveBeenCalled();
  });
});
