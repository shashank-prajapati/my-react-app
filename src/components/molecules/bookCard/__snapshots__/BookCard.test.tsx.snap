// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Book Card should match the Buttoned Book Card Snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
      data-testid="Card"
    >
      <img
        alt="Book"
        height="100%"
        src="coverImage.png"
        style="display: block;"
        width="100%"
      />
      <div>
        <div
          class="MuiBox-root css-9r861l"
        >
          <div
            class="MuiBox-root css-3jmp7d"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
            >
              Beyond Entrepreneurship
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
            >
              Jim Collins & Bill Lazier
            </p>
            <div>
              <div
                class="MuiBox-root css-5as2nd"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                  focusable="false"
                >
                  clockIcon.svg
                </svg>
                <span
                  class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                >
                  13
                  -minute read
                </span>
              </div>
            </div>
            <div
              class="MuiBox-root css-eb3voy"
            />
          </div>
          <div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Add to library
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Book Card should match the unButtoned Book Card Snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1wzpskv-MuiPaper-root-MuiCard-root"
      data-testid="Card"
    >
      <img
        alt="Book"
        height="100%"
        src="coverImage.png"
        style="display: block;"
        width="100%"
      />
      <div>
        <div
          class="MuiBox-root css-9r861l"
        >
          <div
            class="MuiBox-root css-3jmp7d"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
            >
              Beyond Entrepreneurship
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
            >
              Jim Collins & Bill Lazier
            </p>
            <div>
              <div
                class="MuiBox-root css-5as2nd"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                  focusable="false"
                >
                  clockIcon.svg
                </svg>
                <span
                  class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                >
                  13
                  -minute read
                </span>
              </div>
            </div>
            <div
              class="MuiBox-root css-eb3voy"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r43foo-MuiSvgIcon-root"
                focusable="false"
              >
                horizontalMenu.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Book Card should mouseEnter Snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
      data-testid="Card"
    >
      <img
        alt="Book"
        height="100%"
        src="coverImage.png"
        style="display: block;"
        width="100%"
      />
      <div>
        <div
          class="MuiBox-root css-je0zv2"
        >
          <div
            class="MuiBox-root css-3jmp7d"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
            >
              Beyond Entrepreneurship
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
            >
              Jim Collins & Bill Lazier
            </p>
            <div>
              <div
                class="MuiBox-root css-5as2nd"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                  focusable="false"
                >
                  clockIcon.svg
                </svg>
                <span
                  class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                >
                  13
                  -minute read
                </span>
              </div>
            </div>
            <div
              class="MuiBox-root css-eb3voy"
            />
          </div>
          <div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-47f4e9-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Add to library
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Book Card should mouseLeave Snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vlde2p-MuiPaper-root-MuiCard-root"
      data-testid="Card"
    >
      <img
        alt="Book"
        height="100%"
        src="coverImage.png"
        style="display: block;"
        width="100%"
      />
      <div>
        <div
          class="MuiBox-root css-9r861l"
        >
          <div
            class="MuiBox-root css-3jmp7d"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-i5c7wg-MuiTypography-root"
            >
              Beyond Entrepreneurship
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 css-hi830k-MuiTypography-root"
            >
              Jim Collins & Bill Lazier
            </p>
            <div>
              <div
                class="MuiBox-root css-5as2nd"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2v1qra-MuiSvgIcon-root"
                  focusable="false"
                >
                  clockIcon.svg
                </svg>
                <span
                  class="MuiTypography-root MuiTypography-caption3 css-op2c4e-MuiTypography-root"
                >
                  13
                  -minute read
                </span>
              </div>
            </div>
            <div
              class="MuiBox-root css-eb3voy"
            />
          </div>
          <div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Add to library
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
